crucible/src/Lang/Crucible/CFG/Core.hs

```diff
@@ -518,6 +534,193 @@ instance ExtendContext (TermStmt blocks ret) where
       extC  = extendContext  diff
 
 
+------------------------------------------------------------------------------------
+-- Instantiation / type substitution
+
```

Comment:
I suspect that a lot of this instantiation stuff can be reorganized to use the `traverseFC` methods, which would reduce significantly the amount of traversal boilerplate.

Reply:
That would be nice, but traverseFC does not have the right signature for this function. There's boilerplate here, and it would be good to automate it, but it will probably need its own TemplateHaskell to do so.